### PR TITLE
ref #218 update tutteli-spek-extensions to 1.0.1

### LINKED ISSUE #218: update tutteli-spek-extensions to 1.0.1
*Platform* (jvm, js, android): jvm
*Extension* (none, kotlin 1.3, jdk8): jdk8

## Code related feature

- update spekExtensions_versions to 1.0.1 in build.gradle in the root
- fix PathAssertionsSpec and PathFeatureAssertionsSpec
- compile the project with `./gr build` to verify that all problems have been addressed

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over. 
  This way we get the chance to revise the description in case things have changed in the meantime,
  we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/master/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.  
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](https://slack.kotlinlang.org/) in case you do not have an account yet).

### PR DESCRIPTION
**Pull request by ShradhaSangtani**
* resolves #218 
----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt [hunk 1]
```diff
@@ -29,8 +29,7 @@ import java.nio.file.Path
 import java.nio.file.Paths
 
 object SymbolicLinkResolvingSpec : Spek({
-    val tempFolder = TempFolder.perTest()
-    registerListener(tempFolder)
```
Comment (ShradhaSangtani) [inline domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt]: @robstoll Can you help me with registerListener(). It was not accepting tempFolder as parameter.

**Comment by ShradhaSangtani** [comment]
Also, i am not able to understand why test cases are failing. **java.lang.NoSuchMethodError: 'void org.spekframework.spek2.lifecycle.LifecycleListener.afterExecuteGroup(org.spekframework.spek2.lifecycle.GroupScope)'**

File: domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt (hunk 1 above)
Comment (robstoll) [reply domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt]: You don't require it anymore, `val tempFolder by memoizedTempFolder()` is good enough it registers itself

Comment (ShradhaSangtani) [reply domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt]: Ok got it. I was assuming that is the reason for failing test cases.

Comment (robstoll) [reply domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt]: I am investigating why they fail right now. Locally it builds fine so maybe the problem has to do with skipping tests. Stay tuned

**Comment by robstoll** [comment]
I guess I know why, please update spek to 2.0.8 -- there was a breaking change in spek introduced in version 2.0.8

**Comment by ShradhaSangtani** [comment]
@robstoll Yes, i am getting the tempFolder error. Can you guide me where I need to change exactly? I tried doing this -    ` fun Suite.it(description: String, skip: Skip = No, timeout: Long = delegate.defaultTimeout,tempFolder: MemoizedTempFolder) ` but still getting same error

**Comment by robstoll** [comment]
It's code I have not written myself, thus I have to make a guess without looking into the code again (writting from my mobile). I guess the custom `it` does not delegate to Spek's `it` which means it does not create a test. Try to wrap the body of the custom `it` into Speks `it`. You can also rename the custom `it` to avoid naming conflicts. Maybe @jGleitz has another hint (author of this code)

**Comment by ShradhaSangtani** [comment]
By custom it do you mean Suite.it?
```
 fun Suite.it(description: String, skip: Skip = No, timeout: Long = delegate.defaultTimeout) =
        SymlinkTestBuilder(tempFolder, skipWithLink = ifSymlinksNotSupported) { prefix, innerSkip, body ->
            val skipToUse = if (skip == No) innerSkip else skip
            it(prefix + description, skipToUse, timeout, body)
        }
```

Here, I can see Spek's it is called within custom it. So, Suite.it should call it and create test.

**Comment by robstoll** [comment]
Yes exactly, with custom `it` I meant the custom `Suite.it` functions. OK, my guess was only half correct :D However, I think the hint is still valid. I have now access to the code, so I can help you better. I'll write my hint as review, is probably clearer

**Comment by ShradhaSangtani** [comment]
I tried doing lot of stuff here. I get the reason why tempFolder is not accessible. 
I am trying to do something like this -
```
    fun Suite.it(description: String, skip: Skip = No, timeout: Long = delegate.defaultTimeout) =
        it(
            description, skip, timeout
        ) {
            SymlinkTestBuilder(tempFolder, skipWithLink = ifSymlinksNotSupported){
            }
        }
```
Wrapping the SymlinkTestBuilder creation inside body of Spek it, so that tempFolder is accessible. But I am still confused here. What will be called in lambda of **SymlinkTestBuilder** now? And how will the other **it** methods call withAndWithoutSymlink on SymlinkTestBuilder ?

**Comment by robstoll** [comment]
I'll take a deeper look, I guess you don't have to change the Suite.it but the SymlinkTestBuilder. But give me a moment, I'll look at it now in more depth

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt [hunk 2]
```diff
@@ -77,14 +77,14 @@ abstract class PathAssertionsSpec(
                     every {
                         readAttributes(any(), any<Class<BasicFileAttributes>>(), *anyVararg())
                     } throws IOException(TEST_IO_EXCEPTION_MESSAGE)
-                    every { checkAccess(any(), *anyVararg()) } throws IOException(TEST_IO_EXCEPTION_MESSAGE)
+                    every { checkAccess(any(), *anyVararg()) }throws IOException(TEST_IO_EXCEPTION_MESSAGE)
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: detail but this change is not required

Comment (ShradhaSangtani) [reply misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: reverted. Added by mistake

**Comment by ShradhaSangtani** [comment]
@robstoll Thank you for guiding me. Lots to learn :)

**Comment by ShradhaSangtani** [comment]
The build is failing .Looking into it

**Comment by robstoll** [comment]
@ShradhaSangtani I had a quick look into the build error as well. You have to change CollectingAssertionContainerSpec, line 45 rewrite:
```
 it("returns the assertion when calling ${testee::getAssertions.name} a second time") {
```
to
```
it("returns the assertion when calling ${CollectingAssertionContainer<Int>::getAssertions.name} a second time") {
```
I think that should solve the problem

**Comment by ShradhaSangtani** [comment]
@robstoll Woah. I would have never been able to find the issue. Can you explain in short how this change fixed the build? Also my local build is still failing. The test cases in PathAssertionsSpec is failing. Like - **POSIX: prints parent permission error details** So I was trying to fix those.

**Comment by robstoll** [comment]
It's similar to the error in PathAssertions. Seems like spek introduced the check, that a memoized can only be used within an `it`, in 2.0.8. testee inside the description of `it` was accessed outside of the body of `it`.

**Comment by robstoll** [comment]
@ShradhaSangtani LGTM, I am not going to merge as you have marked this PR as WIP. Let me know once you are ready

**Comment by ShradhaSangtani** [comment]
Got it. I have removed WIP. You can merge it. I just have a doubt. My local build is still failing for PathAssertionSpec with following error -
`
expect the thrown exception: ch.tutteli.atrium.reporting.AtriumError
◆ ▶ message: "expect: /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/__linkTo_nonExistingFile        (sun.nio.fs.UnixPath <3367411>)
◆ to: exist
    » followed the symbolic link /var to /private/var
      ❗❗ found a symbolic link loop: /private/var -> /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/nonExistingFile -> /private/var
    » the closest existing parent directory is /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037"        <1338607038>
    ◾ contains: 
      ⚬ value: "followed the symbolic link /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/__linkTo_nonExistingFile to /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/nonExistingFile"        <1424902477>
        ⚬ ▶ number of occurrences: 0
            ◾ is at least: 1
ch.tutteli.atrium.reporting.AtriumError: expect the thrown exception: ch.tutteli.atrium.reporting.AtriumError
◆ ▶ message: "expect: /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/__linkTo_nonExistingFile        (sun.nio.fs.UnixPath <3367411>)
◆ to: exist
    » followed the symbolic link /var to /private/var
      ❗❗ found a symbolic link loop: /private/var -> /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/nonExistingFile -> /private/var
    » the closest existing parent directory is /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037"        <1338607038>
    ◾ contains: 
      ⚬ value: "followed the symbolic link /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/__linkTo_nonExistingFile to /var/folders/9q/wq6qzxnj0wv_cp0kvvhjg1396zkwtz/T/spek7324942732568221037/nonExistingFile"        <1424902477>
        ⚬ ▶ number of occurrences: 0
            ◾ is at least: 1
	at ch.tutteli.atrium.reporting.AtriumErrorKt.createAtriumError(AtriumError.kt:26)
	at ch.tutteli.atrium.reporting.AtriumError$Companion.create(AtriumError.kt:22)
	at ch.tutteli.atrium.core.robstoll.lib.checking.ThrowingAssertionChecker.check(ThrowingAssertionChecker.kt:43)
	at ch.tutteli.atrium.creating.ReportingAssertionContainer$EagerCommonFields.check(ReportingAssertionContainer.kt:111)
	at ch.tutteli.atrium.core.robstoll.lib.creating.ReportingAssertionContainerImpl.checkAndClearAssertions(ReportingAssertionContainerImpl.kt:35)
	at ch.tutteli.atrium.core.robstoll.lib.creating.ReportingAssertionContainerImpl.addAssertion(ReportingAssertionContainerImpl.kt:29)
	at ch.tutteli.atrium.core.robstoll.lib.creating.ReportingAssertionContainerImpl.addAssertion(ReportingAssertionContainerImpl.kt:11)
	at ch.tutteli.atrium.core.robstoll.lib.checking.DelegatingAssertionChecker.check(DelegatingAssertionChecker.kt:13)
	at ch.tutteli.atrium.creating.ReportingAssertionContainer$EagerCommonFields.check(ReportingAssertionContainer.kt:111)
	at ch.tutteli.atrium.core.robstoll.lib.creating.ReportingAssertionContainerImpl.checkAndClearAssertions(ReportingAssertionContainerImpl.kt:35)
	at ch.tutteli.atrium.core.robstoll.lib.creating.ReportingAssertionContainerImpl.addAssertion(ReportingAssertionContainerImpl.kt:29)
	at ch.tutteli.atrium.domain.creating.changers.PostFinalStep.addToInitial(PostFinalStep.kt:66)
	at ch.tutteli.atrium.api.fluent.en_GB.FeatureAssertionsKt.feature(featureAssertions.kt:36)
	at ch.tutteli.atrium.api.fluent.en_GB.ThrowableAssertionsKt.message(throwableAssertions.kt:82)
	at ch.tutteli.atrium.specs.integration.PathAssertionsSpec$1$8$1$1.invoke(PathAssertionsSpec.kt:185)
	at ch.tutteli.atrium.specs.integration.PathAssertionsSpec$1$8$1$1.invoke(PathAssertionsSpec.kt:31)
	at ch.tutteli.atrium.specs.integration.SymlinkTestBuilder$callWith$1.invoke(PathAssertionsSpec.kt:780)
	at ch.tutteli.atrium.specs.integration.SymlinkTestBuilder$callWith$1.invoke(PathAssertionsSpec.kt:753)
	at org.spekframework.spek2.runtime.scope.TestScopeImpl.execute(scopes.kt:136)
	at org.spekframework.spek2.runtime.Executor$execute$result$2$1$job$1.invokeSuspend(Executor.kt:74)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)`

Why this didn't fail in build.And what could be the reason for this failure?

**Comment by ShradhaSangtani** [comment]
Yes I am using mac OS. Yes maybe something like that. Thanks :)

**Comment by robstoll** [comment]
@jGleitz, another use case, I guess we need to revise the symlink loop detection

**Comment by robstoll** [comment]
@ShradhaSangtani thanks for the contribution 👍 
I encourage you to continue your contributions -- they are always welcome and I help also outside of hacktoberfest. 1 PR to go and you have your hat-trick 😉 
https://github.com/robstoll/atrium/issues?q=is%3Aopen+is%3Aissue+no%3Aassignee+label%3A%22help+wanted%22

**Comment by robstoll** [comment]
@ShradhaSangtani could you do me a favour. Can you please checkout/rebase on master and check if PathAssertionSpec still fails?

**Pull request merged**